libs/coin-evm/src/specs.ts

```diff
@@ -245,71 +249,122 @@ const evmBasicMutations: ({
       });
     },
   },
-  {
-    name: "move some ERC20",
-    maxRun: 1,
-    transaction: ({ account, siblings, bridge }): TransactionRes<EvmTransaction> => {
-      const erc20Account = sample((account.subAccounts || []).filter(a => a.balance.gt(0)));
-      invariant(erc20Account, "no erc20 account");
-      const sibling = pickSiblings(siblings, 3);
-      const recipient = sibling.freshAddress;
+];
+
+const moveErc20Mutation: MutationSpec<EvmTransaction> = {
+  name: "move some ERC20 like (ERC20, BEP20, etc...)",
+  maxRun: 1,
+  transaction: ({ account, siblings, bridge }): TransactionRes<EvmTransaction> => {
+    const erc20Account = sample((account.subAccounts || []).filter(a => a.balance.gt(0)));
+    invariant(erc20Account, "no erc20 account");
+    const sibling = pickSiblings(siblings, 3);
+    const recipient = sibling.freshAddress;
+    return {
+      transaction: bridge.createTransaction(account),
+      updates: [
+        {
+          recipient,
+          subAccountId: erc20Account!.id,
+        },
+        Math.random() < 0.5
+          ? {
+              useAllAmount: true,
+            }
+          : {
+              amount: erc20Account!.balance.times(Math.random()).integerValue(),
+            },
+      ],
+    };
+  },
+  test: ({ accountBeforeTransaction, account, transaction }): void => {
+    invariant(accountBeforeTransaction.subAccounts, "sub accounts before");
+    const erc20accountBefore = accountBeforeTransaction.subAccounts?.find(
+      s => s.id === transaction.subAccountId,
+    );
+    invariant(erc20accountBefore, "erc20 acc was here before");
+    invariant(account.subAccounts, "sub accounts");
+    const erc20account = account.subAccounts!.find(s => s.id === transaction.subAccountId);
+    invariant(erc20account, "erc20 acc is still here");
+
+    if (transaction.useAllAmount) {
+      botTest("erc20 account is empty", () => expect(erc20account!.balance.toString()).toBe("0"));
+    } else {
+      botTest("account balance moved with tx amount", () =>
+        expect(erc20account!.balance.toString()).toBe(
+          erc20accountBefore!.balance.minus(transaction.amount).toString(),
+        ),
+      );
+    }
+  },
+};
+
+const getAppQuery = (currencyId: CryptoCurrency["id"]): AppSpec<EvmTransaction>["appQuery"] => {
+  switch (currencyId) {
+    case "avalanche_c_chain":
+      return { model: DeviceModelId.nanoS, appName: "Avalanche" };
+    case "polygon":
+      return { model: DeviceModelId.nanoS, appName: "Polygon" };
+    case "bsc":
+      return { model: DeviceModelId.nanoS, appName: "Binance Smart Chain" };
+    case "ethereum_classic":
+      return { model: DeviceModelId.nanoS, appName: "Ethereum Classic" };
```

Comment:
Not a blocker for this PR, but it would be great to start using NanoS+ apps for the bot instead of NanoS apps, which are getting more and more deprecated.

Would be cool to test if you have a chance.

Reply:
Does not work with nanos+ at first glance (cf. critical speculos errors in https://github.com/LedgerHQ/ledger-live/pull/4563#issuecomment-1704940864)
Not blocker for this merge since develop still uses nanos. Should be worked on independently